src/sage/libs/flint/fmpz.pxd

```diff
@@ -1,195 +1,1126 @@
 # distutils: libraries = flint
 # distutils: depends = flint/fmpz.h
 
+################################################################################
+# This file is auto-generated. Do not modify by hand
+################################################################################
+
 from libc.stdio cimport FILE
-from sage.libs.gmp.types cimport mpz_t
+from sage.libs.gmp.types cimport *
+from sage.libs.mpfr.types cimport *
 from sage.libs.flint.types cimport *
 
-# flint/fmpz.h
 cdef extern from "flint_wrap.h":
-    # Memory management
-    void fmpz_init(fmpz_t)
-    void fmpz_init2(fmpz_t, ulong limbs)
 
-    void fmpz_clear(fmpz_t)
+    fmpz PTR_TO_COEFF(__mpz_struct * ptr) noexcept
+
+    __mpz_struct * COEFF_TO_PTR(fmpz f) noexcept
+
+    int COEFF_IS_MPZ(fmpz f) noexcept
```

Comment:
conflicts with previous declaration in `types.pxd`

Reply:
Indeed.`COEFF_IS_MPZ` is a macro in `flint/fmpz.h`. However, in `flint/doc/source/fmpz.rst` it is declared as a function
```rest
.. function:: int COEFF_IS_MPZ(fmpz f)
```
and hence automatically declared by the autogeneration script. It ought to be a `.. macro` in `flint/doc/source/fmpz.rst` and be in `fmpz_macros.pxd` rather than `types.pxd`.